test/conftest.py

```diff
@@ -342,9 +334,7 @@ def configure_paths_for_tests(tmp_path_factory, request):
     gha = os.environ.get("GITHUB_ACTIONS", False)
     # Under what circumstances do we want to use a temporary input directory?
     # This will force a re-download of raw inputs from Zenodo or the GCS cache:
-    if (gha and "PUDL_INPUT" not in os.environ) or (
-        request.config.getoption("--tmp-data")
-    ):
+    if request.config.getoption("--tmp-data") or ("PUDL_INPUT" not in os.environ):
```

Comment:
I'm not 100% sure if/how this is related but, because we cache the inputs on GHA to avoid downloading them over and over again, they need to be in a fixed location on the runner and can't be in a randomly generated temporary directory (so the caching mechanism knows where to put the files, which happens before we start the tests)

IIRC there might also be some special cases where we *always* want to use a temporary directory e.g. to force a minimal test of the downloading facility even if we have cached inputs.

Reply:
I think if we set `PUDL_INPUT` on GHA (which I think we do) then the caching at least should be unaffected by this change.

Reply:
Hmph, some tests were failing locally for me, I suppose `PUDL_INPUT` was not set then? I assume that we should always have this configured, and we don't expect to ever hit a scenario where this shouldn't be set, right? Maybe we should proactively abort here if the env variables are not set.

Reply:
I think we should:
* remove this `gha and PUDL_INPUT not set` check, since that "should" never happen
* add a defensive check for PUDL_INPUT